### PR TITLE
Feature/sinf 372 spree iam user

### PR DESCRIPTION
**Pull request by adrianmilne**
This update will remove the need to manually create the `spree-user`, set permissions and access key system parameters during install.

The reasons for doing this are:
1. Tighten security on that user
2. Make it more consistently reproducible
3. Remove manual steps (and therefore make OAT/handover easier)

The Terraform has been updated to create a new IAM user `spree-app-user` (new name so it can exist alongside the old one for now). It will add a policy to grant access to the spree S3 buckets only (I assume this will sufficient - it had full access on all S3 buckets previously), create an access key and associated system parameters and inject those into the ECS tasks.

I have also added some documentation to show how to rotate the access key in future if/when needed (to ensure it stays under Terraform control - thanks to @tommyb82 for the steer).

https://crowncommercialservice.atlassian.net/wiki/spaces/SCALE/pages/2422439947/Rotate+spree-app-user+IAM+access+key

### CODE REVIEW

File: terraform/modules/configs/deploy-all/main.tf
Comment (adrianmilne) [inline terraform/modules/configs/deploy-all/main.tf]: Thought there may be a more elegant way to pass these as an array - but wasn't sure

File: terraform/modules/iam/main.tf
Comment (tommyb82) [inline terraform/modules/iam/main.tf]: Need to encrypt these two using the new CMK - see example in previous PR: https://github.com/Crown-Commercial-Service/ccs-scale-infra-services-fat/pull/170/files#diff-414d0fcf4cadd0901a7bc0ec77e1bb1d265d51d2ee55f84da485cbeef55d677cR93-R102

File: terraform/modules/configs/deploy-all/main.tf
Comment (tommyb82) [reply terraform/modules/configs/deploy-all/main.tf]: Yep I would have thought passing in a `list` type var using the built-in `[obj1,obj1...]` syntax should work - and be compatible with the resource list in the policy statement?

File: terraform/modules/iam/main.tf
Comment (tommyb82) [inline terraform/modules/iam/main.tf:42]: I like this new way of defining the policy statements, much clearer than the old heredoc style 👍

**Review by tommyb82 — CHANGES_REQUESTED** [review]
Excellent - just the one consequential change required I think 👍

File: terraform/modules/iam/main.tf
Comment (adrianmilne) [reply terraform/modules/iam/main.tf]: Great point - forgot to update after that change - will do so now 👍

File: terraform/modules/configs/deploy-all/main.tf
Comment (adrianmilne) [reply terraform/modules/configs/deploy-all/main.tf]: The bit I wasn't sure about was appending the `/*` suffix to the arns for this case (assuming a list of arns was passed in)

```
Resource = [
          "${var.s3_static_bucket_arn}/*",
          "${var.s3_feed_bucket_arn}/*",
          "${var.s3_cnet_bucket_arn}/*",
          "${var.s3_product_import_bucket_arn}/*"
        ]
```

Comment (tommyb82) [reply terraform/modules/configs/deploy-all/main.tf]: Ahh yep, missed that!  `formatlist` might help there, whether it's worth it or not.. 🤷‍♂️ 
https://www.terraform.io/docs/language/functions/formatlist.html

**Review by tommyb82 — APPROVED** [review]
Awesome!

File: terraform/modules/configs/deploy-all/main.tf
Comment (adrianmilne) [reply terraform/modules/configs/deploy-all/main.tf]: Missed this in the chaos - good call - just applied that now and it seems to work 👍  Will need to rebase after the other S3 tidy up PR (SINF-394) is merged so will hold off merging until then

**Pull request merged**
